weather-subscription-service/internal/mailer/manager.go

```diff
@@ -0,0 +1,273 @@
+package mailer
+
+import (
+	"context"
+	"pkg/protos/mailer"
+	"sync"
+	"time"
+	"weather-subscription/internal/config"
+	"weather-subscription/internal/models"
+	"weather-subscription/internal/weather"
+	"weather-subscription/pkg/mailservice"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	Day                    = 24 * time.Hour
+	SendEmailDailyTimeout  = time.Minute * 15
+	SendEmailHourlyTimeout = time.Minute * 15
+	SendEmailTimeout       = time.Minute * 16
+	LoadTimeoutDuration    = time.Second * 5
+	MaxRetries             = 3
+	RetryDelay             = time.Second * 2
+)
+
+type MailerStore interface {
+	GetSubscribed(ctx context.Context) ([]models.Subscription, error)
+}
+
+type Manager struct {
+	MailService  *mailservice.MailService
+	Targets      *TargetManager
+	Forecasts    *Forecaster
+	Logger       logger
+	emailBuilder *EmailBuilder
+
+	stopChan chan struct{}
+	wg       sync.WaitGroup
+	running  bool
+}
+
+type logger interface {
+	ConsoleLogInfo(msg string, fields ...zap.Field)
+	LogInfo(msg string, fields ...zap.Field)
+	ConsoleLogError(msg string, fields ...zap.Field)
+	LogError(msg string, fields ...zap.Field)
+}
+
+func New(grpcConfig config.GRPCConfig, weatherAPIService *weather.WeatherAPIService, logger logger) *Manager {
+	forecaster := NewForecaster(weatherAPIService)
```

Comment:
```suggestion
	forecaster := NewForecaster(weatherAPIService)
	mailService := mailservice.NewMailService(grpcConfig, logger)
	emailBuilder := NewEmailBuilder()
```

Why not init them in `main.go` as dependencies?

Reply:
when I created this package I did everything interfaces and it seems quite like interface polution, so I decided to have concrete realization, so because I create objects It is have no difference - create in main and pass or create here

Reply:
ok, resolved